### DISCUSSION TITLE
Self-hosted runners performance

### DISCUSSION THREAD

**Discussion by joshua9519** [community/community#31352] [Actions]
Hi everyone,
I have recently deployed Github self-hosted runners (using the Github [Action Runners Controller](https://github.com/actions-runner-controller/actions-runner-controller)) to GKE. This was done to have more control over the performance of the runners and hopefully allow us to speed up our workflows.
I did some reading and found that Github-hosted runners have:
- 2 cores CPU
- 7GB RAM

I have tested with one of our workflows which runs some tests and with Github-hosted runners, it takes around 8 and a half minutes. When I gave our runners the same resource requests as the Github runners and ran the same workflow, it took over 11 minutes and needs over twice the resources to match the time.

Does anyone have any pointers on improving the performance of the runners when hosted on Kubernetes?

Thanks!

**Comment by vitalykarasik** [comment]
I suggest 'profile' your job, i.e. to compare how long each step is taking.
At a glance there are at least  two potential explanations:
1) different VM specs in Azure and GCP. In general, it's very hard to compare "CPU" in the cloud. Plus storage performance.
2) managed runners are close to Github servers, so checkout may be faster

**Reply by joshua9519** [reply]
Hi Vitaly, thanks for the quick response! I have tried to profile the jobs and generally everything seems to take longer unfortunately. I'll continue to look into it.
